course6/toyapp_ariel/displaylib/mat4.h

```diff
@@ -0,0 +1,84 @@
+﻿#pragma once
+
+#include <cassert>
```

Comment:
Always order your includes from the most local ones, to the most general ones; so start at the local class' header, and finish with STL and libc headers:
```
#include "float3.h"
#include "float4.h"

#include <SDL2/SDL_main.h> // why, on earth, am I including this here?

#include <cassert>
```